apps/els_lsp/src/els_completion_provider.erl

```diff
@@ -183,12 +209,188 @@ find_completions( Prefix
+-spec paths_include(els_dt_document:item()) -> [binary()].
+paths_include(#{uri := Uri}) ->
+  case match_in_path(els_uri:path(Uri), els_config:get(apps_paths)) of
+    [] ->
+      [];
+    [Path|_] ->
+      AppPath = filename:join(lists:droplast(filename:split(Path))),
+      {ok, Headers} = els_dt_document_index:find_by_kind(header),
+      lists:flatmap(
+        fun(#{uri := HeaderUri}) ->
+            case string:prefix(els_uri:path(HeaderUri), AppPath) of
+              nomatch ->
+                [];
+              IncludePath ->
+                [relative_include_path(IncludePath)]
+            end
+        end, Headers)
+  end.
+
+-spec paths_include_lib() -> [binary()].
+paths_include_lib() ->
+  Paths = els_config:get(otp_paths)
+    ++ els_config:get(deps_paths)
+    ++ els_config:get(apps_paths)
+    ++ els_config:get(include_paths),
+  {ok, Headers} = els_dt_document_index:find_by_kind(header),
+  lists:flatmap(
+    fun(#{uri := Uri}) ->
+        HeaderPath = els_uri:path(Uri),
+        case match_in_path(HeaderPath, Paths) of
+          [] ->
+            [];
+          [Path|_] ->
+            <<"/", PathSuffix/binary>> = string:prefix(HeaderPath, Path),
+            PathBin = unicode:characters_to_binary(Path),
+            case lists:reverse(filename:split(PathBin)) of
+              [<<"include">>, App | _] ->
+                [filename:join([ strip_app_version(App)
+                               , <<"include">>
+                               , PathSuffix])];
+              _ ->
+                []
+            end
+        end
+    end, Headers).
+
+-spec match_in_path(binary(), [binary()]) -> [binary()].
+match_in_path(DocumentPath, Paths) ->
+  [P || P <- Paths, string:prefix(DocumentPath, P) =/= nomatch].
+
+-spec relative_include_path(binary()) -> binary().
+relative_include_path(Path) ->
+  case filename:split(Path) of
+    [_App, <<"include">> | Rest] -> filename:join(Rest);
+    [_App, <<"src">> | Rest]     -> filename:join(Rest);
+    [_App, SubDir | Rest]        -> filename:join([<<"..">>, SubDir|Rest])
+  end.
+
+-spec strip_app_version(binary()) -> binary().
+strip_app_version(App0) ->
+  %% Transform "foo-1.0" into "foo"
+  case string:lexemes(App0, "-") of
+    []  -> App0;
+    [_] -> App0;
+    Lexemes ->
+      Vsn = lists:last(Lexemes),
+      case re:run(Vsn, "^[0-9.]+$", [global, {capture, none}]) of
+        match   -> list_to_binary(lists:join("-", lists:droplast(Lexemes)));
+        nomatch -> App0
+      end
+  end.
+
+-spec item_kind_file(binary()) -> item().
+item_kind_file(Path) ->
+  #{ label            => Path
+   , kind             => ?COMPLETION_ITEM_KIND_FILE
+   , insertTextFormat => ?INSERT_TEXT_FORMAT_PLAIN_TEXT
+   }.
+
+%%==============================================================================
+%% Snippets
+%%==============================================================================
+-spec snippet(atom()) -> item().
+snippet(attribute_behaviour) ->
```

Comment:
Should we use the snippet server for this? (can be done as a follow up). Look at `els_snippets_server` and `apps/els_lsp/priv/snippets/`

Reply:
I guess so, I will look into it.
I'm not too familiar with the `els_snippets_server`, what would the advantage of using that be?

Reply:
Mainly to have all snippets in one place and not hard-coded in code. Snippets are loaded from a directory, which means they can be overwritten by the user (e.g. if you prefer a slightly different formatting than the default ones).

Reply:
I will look into it!